arkouda/strings.py

```diff
@@ -255,6 +260,146 @@ def get_lengths(self) -> pdarray:
+        """
+        Prints the regex patterns for which Strings.find(pattern) has been cached
+        """
+        self._refresh_regex_dict()
+        print(self.regex_dict.keys())
+
+    @typechecked
+    def find(self, pattern: Union[bytes, str_scalars]) -> Tuple[pdarray, pdarray, pdarray]:
+        """
+        Finds pattern matches and returns pdarrays containing the number, start postitions, and lengths of matches
+            Note: only handles regular expressions supported by re2 (does not support lookaheads/lookbehinds)
+
+        Parameters
+        ----------
+        pattern: str_scalars
+            The regex pattern used to find matches
+
+        Returns
+        -------
+        pdarray, int64
+            For each original string, the number of pattern matches
+        pdarray, int64
+            The start positons of pattern matches
+        pdarray, int64
+            The lengths of pattern matches
+
+        Raises
+        ------
+        TypeError
+            Raised if the pattern parameter is not bytes or str_scalars
+        ValueError
+            Rasied if pattern is not a valid regex
+        RuntimeError
+            Raised if there is a server-side error thrown
+
+        See Also
+        --------
+        Strings.slice, Strings.match
+
+        Examples
+        --------
+        >>> strings = ak.array(['{} string {}'.format(i, i) for i in range(1, 6)])
+        >>> strings
+        array(['1 string 1', '2 string 2', '3 string 3', '4 string 4', '5 string 5'])
+        >>> num_matches, starts, lens = strings.find('\\d')
+        >>> num_matches
+        array([2, 2, 2, 2, 2])
+        >>> starts
+        array([0, 9, 11, 20, 22, 31, 33, 42, 44, 53])
+        >>> lens
+        array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1]))
+        """
+        if isinstance(pattern, bytes):
+            pattern = pattern.decode()
+        sym_tab = list_symbol_table()
+        if pattern not in self.regex_dict or any([pda.name not in sym_tab for pda in self.regex_dict[pattern]]):
+            # run find if we don't have the result of find(pattern) cached or any of the references have gone stale
+            try:
+                re.compile(pattern)
+            except Exception as e:
+                raise ValueError(e)
+            cmd = "segmentedFind"
+            args = "{} {} {} {}".format(self.objtype,
+                                        self.offsets.name,
+                                        self.bytes.name,
+                                        json.dumps([pattern]))
+            repMsg = cast(str, generic_msg(cmd=cmd, args=args))
+            arrays = repMsg.split('+', maxsplit=2)
+            self.regex_dict[pattern] = (create_pdarray(arrays[0]), create_pdarray(arrays[1]), create_pdarray(arrays[2]))
+        return self.regex_dict[pattern]
+
+    @typechecked
+    def slice(self, pattern: Union[bytes, str_scalars], return_match_origins: bool = False) -> Union[Strings, Tuple]:
```

Comment:
I agree with @reuster986 on the naming here.  I'd try to make the naming match the `re2` naming of what's happening here if possible.  `slice` seems to imply a different context..something you apply to the structure of an array/list.  Looking at the ongoing conversation it sounds like you are already going to change it.